client/src/routes/profile.js

```diff
@@ -184,51 +204,81 @@ const UpdateCard = withStyles({
     },
     padding: '0 !important'
   }
-}, ({ name, divisionId, onUpdate, classes }) => {
+}, ({ name: oldName, email: oldEmail, onUpdate, classes }) => {
   const { toast } = useToast()
 
   const [deleteModalVisible, setDeleteModalVisible] = useState(false)
   const dismissDeleteModal = useCallback(() => setDeleteModalVisible(false), [])
   const handleDelete = useCallback(() => setDeleteModalVisible(true), [])
 
-  const [updateName, setUpdateName] = useState(name)
-  const handleUpdateName = useCallback((e) => setUpdateName(e.target.value), [])
+  const [name, setName] = useState(oldName)
+  const handleSetName = useCallback((e) => setName(e.target.value), [])
+
+  const [email, setEmail] = useState(oldEmail)
+  const handleSetEmail = useCallback(e => setEmail(e.target.value), [])
 
   const [isButtonDisabled, setIsButtonDisabled] = useState(false)
 
   const doUpdate = useCallback((e) => {
     e.preventDefault()
 
-    setIsButtonDisabled(true)
+    if (name !== oldName) {
+      setIsButtonDisabled(true)
+      updateAccount({
+        name
+      })
+        .then(({ error, data }) => {
+          setIsButtonDisabled(false)
+
+          if (error !== undefined) {
+            toast({ body: error, type: 'error' })
+            return
+          }
+
+          toast({ body: 'Profile updated' })
+
+          onUpdate({
+            name: data.user.name,
+            divisionId: Number.parseInt(data.user.division)
+          })
+        })
+    }
+
+    if (email !== oldEmail) {
+      setIsButtonDisabled(true)
 
-    updateAccount({
-      name: updateName
-    })
-      .then(({ error, data }) => {
+      const handleResponse = ({ error, data }) => {
         setIsButtonDisabled(false)
 
         if (error !== undefined) {
           toast({ body: error, type: 'error' })
           return
         }
 
-        toast({ body: 'Profile updated' })
+        toast({ body: data })
+      }
 
-        onUpdate({
-          name: data.user.name,
-          divisionId: Number.parseInt(data.user.division)
+      if (email === '') {
+        deleteEmail()
+          .then(handleResponse)
+      } else {
+        updateEmail({
+          email
         })
-      })
-  }, [updateName, onUpdate, toast])
+          .then(handleResponse)
+      }
+    }
+  }, [name, email, oldName, oldEmail, onUpdate, toast])
 
   return (
     <div class='card u-flex u-flex-column'>
       <div class='content'>
         <p>Update Information</p>
-        <p class='font-thin u-no-margin'>This will change how your team appears on the scoreboard. Note that you may only update your team's information once every 10 minutes.</p>
+        <p class='font-thin u-no-margin'>This will change how your team appears on the scoreboard. Note that you may only change your team's name once every 10 minutes (changing only email won't count).</p>
```

Comment:
```suggestion
        <p class='font-thin u-no-margin'>This will change how your team appears on the scoreboard. Note that you may only change your team's name once every 10 minutes.</p>
```

imo it's fine without the stuff in parenthesis.